### PR TITLE
Enhancement: reset to original commit after fixup

### PR DESCRIPTION
**Pull request by randy3k**
- reset to original commit after fixup 
- fixup should be safe even if there are untracked files

### CODE REVIEW

File: core/commands/fixup.py [hunk 1]
```diff
@@ -59,6 +59,7 @@ def do_action(self, commit, **kwargs):
 
             self.rewrite_active_branch(base_commit, commit_chain)
         except Exception as e:
+            self.git("reset", "--hard", commit)
```
Comment (stoivo) [inline core/commands/fixup.py]: Why do we want to do this?
If we do a reset --hard commit (This is the commit you want to rebase the changes into??), would we remove the commits from commit unto HEAD?

Comment (randy3k) [reply core/commands/fixup.py]: Ai, we should reset only....reset —hard would be too much. The idea is to get rid of the fixup commit.

Comment (stoivo) [reply core/commands/fixup.py]: `HEAD^1` ?
So one commit before HEAD?

Comment (randy3k) [reply core/commands/fixup.py]: 👍

**Comment by randy3k** [comment]
pushed, please review again.

**Pull request merged**